drivers: regulator: axp192: fix init priority

bbilas opened the pull request:
That was missed in  https://github.com/zephyrproject-rtos/zephyr/commit/66f5fce68fa1c982df1b2639c49446e3e7bf0df9 commit so let's adjust it as well.

bbilas committed:
drivers: regulator: axp192: fix init priority

That was missed in 66f5fce commit so let's adjust it as well.

Signed-off-by: Bartosz Bilas <bartosz.bilas@hotmail.com>

fabiobaltieri commented:
Your hash reference in the commit message is unclear, if you want to refer to the PR use #64072, but if you want to refer to the short hash (which is the right thing to do IMO) then you should use 3ce916d68f, as that's the sha of the rebased commit. Could you update the message?

bbilas commented:
@fabiobaltieri This should have been merged before https://github.com/zephyrproject-rtos/zephyr/pull/64673 but I've forgotten to say that.

fabiobaltieri commented:
> @fabiobaltieri This should have been merged before #64673 but I've forgotten to say that.

Yeah it's alright, happens.

bbilas commented:
> Your hash reference in the commit message is unclear, if you want to refer to the PR use #64072, but if you want to refer to the short hash (which is the right thing to do IMO) then you should use [66f5fce](https://github.com/zephyrproject-rtos/zephyr/commit/66f5fce68fa1c982df1b2639c49446e3e7bf0df9), as that's the sha of the rebased commit. Could you update the message?
> 
> UPDATE: [66f5fce](https://github.com/zephyrproject-rtos/zephyr/commit/66f5fce68fa1c982df1b2639c49446e3e7bf0df9) is the right one

Done.

bbilas commented:
@fabiobaltieri nah, it seems that we need to override that priority for m5stack_core2 to smaller than DISPLAY_INIT_PRIORITY which is 85 because of that failure.

bbilas committed:
boards: m5stack_core2: override REGULATOR_AXP192_INIT_PRIORITY priority

axp192 is used by the display controller and gpio hog subsys
thus we need to set this priority to the smaller value.

Signed-off-by: Bartosz Bilas <bartosz.bilas@hotmail.com>

fabiobaltieri commented:
I'll be honest, I think SPI init priority should have been changed instead, I tried before and got pushed back. But that before doing the build time validation. But now I have other idea so I guess whatever magic combination of number works will do until we come up with something better.

bbilas commented:
> I'll be honest, I think SPI init priority should have been changed instead, I tried before and got pushed back. But that before doing the build time validation. But now I have other idea so I guess whatever magic combination of number works will do until we come up with something better.

Yup, SPI priority seems to be too high compared to the e.g I2C which makes many troubles.

gmarull commented:
> > > Yup, SPI priority seems to be too high compared to the e.g I2C which makes many troubles.
> > 
> > 
> > [f65fa6b#r1018151330](https://github.com/zephyrproject-rtos/zephyr/commit/f65fa6b741122cf010c81c118971cddc7d3b7381#r1018151330) this was the conversation at the time, basically it was probably fine but no one wanted to sign off a potential regression, but we had no tools at the time, may be worth bringing it back up.
> 
> Let's open the Pandora box once again!

Be careful. It can be considered a "breaking change" a "treewide change", or even a "stable API change", in all cases, be prepared to have fun.

bbilas commented:
> > > > Yup, SPI priority seems to be too high compared to the e.g I2C which makes many troubles.
> > > 
> > > 
> > > [f65fa6b#r1018151330](https://github.com/zephyrproject-rtos/zephyr/commit/f65fa6b741122cf010c81c118971cddc7d3b7381#r1018151330) this was the conversation at the time, basically it was probably fine but no one wanted to sign off a potential regression, but we had no tools at the time, may be worth bringing it back up.
> > 
> > 
> > Let's open the Pandora box once again!
> 
> Be careful. It can be considered a "breaking change" a "treewide change", or even a "stable API change", in all cases, be prepared to have fun.

It would be interesting to see the CI results.

fabiobaltieri commented:
> It would be interesting to see the CI results.

Go Twister! https://github.com/zephyrproject-rtos/zephyr-testing/actions/runs/6735192515/job/18307793563

fabiobaltieri commented:
@bbilas seems like the answer is: nothing, at least for the push run https://github.com/zephyrproject-rtos/zephyr-testing/runs/18312769006

bbilas commented:
> @bbilas seems like the answer is: nothing, at least for the push run https://github.com/zephyrproject-rtos/zephyr-testing/runs/18312769006

well, who is gonna sign off? ;) Let's make a PR and try to sort it out.

fabiobaltieri commented:
> well, who is gonna sign off? ;) Let's make a PR and try to sort it out.

Ahhh... your turn! :-)
